crates/core/tedge_mapper/src/c8y/service_monitor.rs

```diff
@@ -0,0 +1,126 @@
+use crate::core::error::ConversionError;
+use c8y_api::smartrest::topic::SMARTREST_PUBLISH_TOPIC;
+use mqtt_channel::Message;
+use mqtt_channel::Topic;
+use serde_json::Value;
+use std::collections::HashMap;
+
+pub fn convert_health_status_message(
+    message: &Message,
+    device_name: String,
+) -> Result<Vec<Message>, ConversionError> {
+    let mut mqtt_messages: Vec<Message> = Vec::new();
+    let topic = message.topic.name.to_owned();
+    let payload: HashMap<String, Value> = serde_json::from_str(message.payload_str()?)?;
```

Comment:
I used a hashmap here because the status payload will have some optional fields. Thin-edge services publish `{"pid"="1234", "time"=" ", "status"="up/down"}` but for service monitor `pid & time` never used. The type is also optional, it will have a default value as `thin-edge.io` if not present.
In the case of child devices, I propose just sending the `status and type` as part of the health status message is enough.

Reply:
The issue is that a HashMap is a weak struct. As a consequence the current code is again and again providing a default for type and status.

Why not simply:

```
#[derive(Deserialize)]
pub struct HealthStatus {
    #[serde(default = "thin-edge.io")]
    pub type: String,

    #[serde(default = "down")]
    pub status: String,
}
```

Reply:
Make sense, refactored the code to use the struct.